### DISCUSSION TITLE
PriorityQueue - How to increase priority?

### DISCUSSION THREAD

**Discussion by Symbai** [dotnet/runtime#52091] [Q&A] [answered]
After adding elements to PriorityQueue, how can I increase the priority of a specific element in this queue?

**Comment by gfoidl** [comment, marked as answer]
> increase the priority of a specific element in this queue

Dequeue and Enqueue.

There's no method for updating priority in-place, as this doesn't play nice with the underlying data-structure. In other words: this would penalize all other operations, that's why it's done this way.

**Reply by Symbai** [reply]
That's a terrible news, as in the discussion of this class there were several API's mentioned for updating the priority. Now that this won't work, it cuts off many use cases.

Dequeue and Enqueue is not an option. If it would be, it would make the priority thing pretty much obsolete. Everyone who wants the normal Queue<T> to deal items by priority you can tell them the same thing. Dequeue and Enqueue until its in the order of the priority.

Anyway, API has been approved this way.

**Reply by gfoidl** [reply]
You can upvote https://github.com/dotnet/runtime/issues/44871

Edit: see you did that already 👍🏻
